packages/babel-core/src/parse.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+
+import loadConfig, { InputOptions } from "./config";
+import normalizeFile from "./transformation/normalize-file";
+import normalizeOptions from "./transformation/normalize-opts";
+
+type AstRoot = BabelNodeFile | BabelNodeProgram;
+
+export default function parse(
+  code: string,
+  opts?: InputOptions,
```

Comment:
Is this the API that we want? Technically, this should require either a `filename` property or the desired configuration, so I suppose it's not really optional.

Reply:
Maybe we should add some warnings at the top of this function to help a user debug if the above requirements are not met.

Reply:
Personally I'd love to change

https://github.com/babel/babel/blob/252ea5a966c1968d8aac21a1a81c6d45173e57dd/packages/babel-core/src/config/config-chain.js#L126

to
```
  if (opts.babelrc !== false) {
    if (context.filename === null) {
      throw new Error("Babel was not told to skip babelrc files, but was given no filename to search from");
    const filename = context.filename;
```

Reply:
On the fence about if that would be too aggressive, but it seems like a case where it would help users, so they'd have to explicitly pass `babelrc: false` if they didn't pass in a filename.

Reply:
Would you like that change to happen in this PR?

Reply:
I'm up for it if it sounds good to you. The only thing I'll say is I think we should be moving toward throwing clearer errors, so I'd vote for probably like

```
  if (context.filename === null) {
    throw Object.assign(new Error("Babel was not told to skip babelrc files, but was given no filename to search from");
  }
```

Reply:
I'm having trouble getting Flow to cooperate here - will leave it for now and try to pick it up in a future PR.